src/game.cpp

```diff
@@ -1701,6 +1701,84 @@ ReturnValue Game::internalTeleport(Thing* thing, const Position& newPos, bool pu
 }
 
 //Implementation of player invoked events
+void Game::playerEquipItem(uint32_t playerId, uint16_t spriteId)
```

Comment:
I think this is close to the simplest it can be, it's a little big because of sanity checks
They guarantee:
- Only move item to backpack if there's space (Can see if I can remove this part if internalMoveItem already does it)
- Only de-equip the item equipped in the proper place (If the sword is in the ammo slot, it won't try to de equip it for example)
- If there's nothing in the slot or if there is and if equipped item is stackable, stack what we have in the backpack with what we have equipped, instead of de-equipping (and return)
- If slot item is equal to equip item, de-equip it, else find item in the backpack, de-equip what's in the slot and equip what's in the backpack

There's two things I can look into this week to see if it's possible to make simpler: Remove backpack capacity sanity check, remove de-equip call before equip. Both in case internalMoveItem already does it.

Sorry if I'm missing something here and it can indeed be made much simpler